sktree/tree/_honest_tree.py

```diff
@@ -0,0 +1,338 @@
+    predict the tied class with the lowest index in :term:`classes_`.
+
+    References
+    ----------
+
+    .. [1] https://en.wikipedia.org/wiki/Decision_tree_learning
+
+    .. [2] L. Breiman, J. Friedman, R. Olshen, and C. Stone, "Classification
+            and Regression Trees", Wadsworth, Belmont, CA, 1984.
+
+    .. [3] T. Hastie, R. Tibshirani and J. Friedman. "Elements of Statistical
+            Learning", Springer, 2009.
+
+    .. [4] L. Breiman, and A. Cutler, "Random Forests",
+            https://www.stat.berkeley.edu/~breiman/RandomForests/cc_home.htm
+
+    .. [5] S. Athey, J. Tibshirani, and S. Wager. "Generalized
+            Random Forests", Annals of Statistics, 2019.
+
+    Examples
+    --------
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.model_selection import cross_val_score
+    >>> from honest_forests import HonestTreeClassifier
+    >>> clf = HonestTreeClassifier(random_state=0)
+    >>> iris = load_iris()
+    >>> cross_val_score(clf, iris.data, iris.target, cv=10)
+    ...                             # doctest: +SKIP
+    ...
+    array([0.93333333, 0.93333333, 1.        , 1.        , 0.93333333,
+           0.8       , 0.8       , 0.93333333, 1.        , 1.        ])
+    """
+
+    def __init__(
+        self,
+        estimator=None,
+        honest_fraction=0.5,
+        honest_prior="empirical",
+    ):
+        if not estimator:
+            self.estimator_ = DecisionTreeClassifier()
+        self.estimator_ = estimator
+        self.honest_fraction = honest_fraction
+        self.honest_prior = honest_prior
+
+    def fit(self, X, y, sample_weight=None, check_input=True):
+        """Build an honest tree classifier from the training set (X, y).
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            The training input samples. Internally, it will be converted to
+            ``dtype=np.float32`` and if a sparse matrix is provided
+            to a sparse ``csc_matrix``.
+
+        y : array-like of shape (n_samples,) or (n_samples, n_outputs)
+            The target values (class labels) as integers or strings.
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Sample weights. If None, then samples are equally weighted. Splits
+            that would create child nodes with net zero or negative weight are
+            ignored while searching for a split in each node. Splits are also
+            ignored if they would result in any single class carrying a
+            negative weight in either child node.
+
+        check_input : bool, default=True
+            Allow to bypass several input checking.
+            Don't use this parameter unless you know what you do.
+
+        Returns
+        -------
+        self : HonestTreeClassifier
+            Fitted estimator.
+        """
+        if check_input:
+            X, y = check_X_y(X, y)
```

Comment:
By calling `estimator_.fit` first, this check is done internally by scikit-learn's baseDecisionTree, so we can remove it.

Reply:
Instead, I would add a check here for `if self.estimator is not of type BaseDecisionTree: <raise some errors>`